### PR TITLE
test(typings): initial tests for typings

### PR DESCRIPTION
**Pull request by layershifter**
We have constant problems with typings (#1609 and more), it's time to solve them ✌️

-----

This PR replaces #1181 and offers initial tests for typings. What it does:
- tests that typings file `Component.d.ts` exists for each component;
- tests that definition file has `ComponentProps` interface and it's exported;
- tests that definition contains all props from `Component.propTypes` and props that are required in definitions are really required.

Also, it fixes all found problems :clinking_glasses: Our typiscript users will be satisfied 👍

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export { default as Flag, FlagProps } from './dist/commonjs/elements/Flag'
 
 export { default as Header, HeaderProps } from './dist/commonjs/elements/Header'
 export { default as HeaderContent, HeaderContentProps } from './dist/commonjs/elements/Header/HeaderContent'
-export { default as HeaderSubheader, HeaderSubHeaderProps } from './dist/commonjs/elements/Header/HeaderSubheader'
+export { default as HeaderSubheader, HeaderSubheaderProps } from './dist/commonjs/elements/Header/HeaderSubheader'
```
Comment (layershifter) [inline index.d.ts:67]: Fixed interface name

File: src/collections/Breadcrumb/BreadcrumbSection.d.ts [hunk 2]
```diff
@@ -15,6 +15,9 @@ export interface BreadcrumbSectionProps {
   /** Additional classes. */
   className?: string;
 
+  /** Shorthand for primary content. */
+  content?: any;
+
```
Comment (layershifter) [inline src/collections/Breadcrumb/BreadcrumbSection.d.ts:20]: Missing definition

File: test/specs/modules/Modal/Modal-test.js [hunk 3]
```diff
@@ -45,6 +45,8 @@ describe('Modal', () => {
   })
 
   common.hasSubComponents(Modal, [ModalHeader, ModalContent, ModalActions, ModalDescription])
+  common.hasValidTypings(Modal)
```
Comment (levithomason) [inline test/specs/modules/Modal/Modal-test.js:48]: Looks like this might already be covered by the `common.isConformant()` test, true?

Comment (layershifter) [reply test/specs/modules/Modal/Modal-test.js:48]: `Modal-test.js` doesn't contain `isConformant` test

Comment (levithomason) [reply test/specs/modules/Modal/Modal-test.js:48]: Of course, thanks!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.68.2`.

**Pull request merged**